src/containers/Allowance/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { FormControl, InputLabel, MenuItem, Select, styled, TextField, Typography } from '@mui/material';
+import { useMemo, useState } from 'react';
+import { formatUnits, isAddress } from 'viem';
+import { useCustomTheme, useTokenList } from '~/hooks';
+import { useToken } from '~/hooks/useToken';
+
+export const Allownace = () => {
+  const tokenList = useTokenList();
+  const { allowance, tokenSelected, selectToken, setTargetAddress } = useToken();
+
+  const [inputAddress, setInputAddress] = useState('');
+
+  const handleChangeToken = (tokenName: string) => {
+    const tokenToBeSelected = tokenList.find((t) => t.tokenData.name === tokenName);
+
+    if (tokenToBeSelected) {
+      selectToken(tokenToBeSelected.tokenData);
+    }
+  };
+
+  const handleChangeInputAddress = (address: string) => {
+    setInputAddress(address);
+
+    if (isAddress(address)) {
+      setTargetAddress(address);
+    }
+  };
+
+  const isValidAddress = useMemo(() => isAddress(inputAddress), [inputAddress]);
+
+  return (
+    <Card>
+      <FormControl fullWidth>
+        <InputLabel id='select-token'>Select Token</InputLabel>
+        <Select
+          labelId='select-token'
+          defaultValue={tokenSelected?.name || ''}
+          value={tokenSelected?.name || ''}
+          label='Select Token'
+          onChange={(e) => handleChangeToken(e.target.value)}
+        >
+          {tokenList.map((t) => (
+            <MenuItem key={`${t.tokenData.chainId}-${t.tokenData.address}`} value={t.tokenData.name}>
```

Comment:
Would add a placeholder option of 'Select token'

Reply:
<img width="291" alt="Screenshot 2024-08-21 at 12 03 42 PM" src="https://github.com/user-attachments/assets/0ef546e9-8441-4a52-bd4c-309061915f1c">

The InputLabel works as a placeholder if there are no tokens and doesn't add an extra option.
Is this ok or am I missing your point?

Reply:
This is perfect!